### PR TITLE
new language: add syphon grammar

### PR DESCRIPTION
**Pull request by yhyadev**
Hey, [Syphon](https://github.com/syphon-lang/syphon) is a language I currently develop and maintain, made a tree sitter parser for developer experience and maintaining it myself instead of somebody else doing it.. now I am trying to add it to your editor, how lovely.

Anyways, I tested it with .config/runtime/queries before I make this pull request but I didn't try it with the current master branch because my pc is bad enough to not build it, I hope you test it before merging.

### CODE REVIEW

**Comment by yhyadev** [comment]
Hmmm, the test suite failed because I didn't run cargo xtask docgen

**Comment by yhyadev** [comment]
Made the change manually with the patch I got from the test suite, Couldn't run `cargo xtask docgen` because of my old 2Gb ram pc

File: runtime/queries/syphon/highlights.scm
Comment (the-mikedavis) [inline runtime/queries/syphon/highlights.scm:1]: Helix uses a custom set of scopes that are different from what neovim uses - https://docs.helix-editor.com/master/themes.html#syntax-highlighting

**Review by the-mikedavis — COMMENTED** [review]
We don't have hard rules about which languages we accept/reject but we try to avoid very small or personal-project languages. Unlike nvim we don't have a way to lazily install parsers so every parser we add increases the size of the release. An alternative to adding the language here is to have instructions in the tree-sitter-syphon repo that show how to add this block to your `~/.config/helix/languages.toml`, call `hx -g fetch` and `hx -g build` and copy the queries into `~/.config/helix/runtime/queries/syphon/*.scm`. Then the queries can be maintained in `queries/helix/*.scm` in the tree-sitter-syphon repo

**Comment by yhyadev** [comment]
Okay, I think I can just maintain it locally then, an thanks for the review I didn't realize you used different scopes

**Pull request closed without merging**